allennlp/training/trainer.py

```diff
@@ -294,7 +302,8 @@ def __init__(
         num_gradient_accumulation_steps: int = 1,
         use_amp: bool = False,
         enable_default_callbacks: bool = True,
-        run_sanity_checks: bool = True,
+        run_sanity_checks: Optional[bool] = None,
```

Comment:
Maybe it's better to just check for this in `**kwargs`? Either way works, I just thought we had gone with the "kwargs" approach elsewhere. But I don't feel strongly either way.

Reply:
Makes sense. I like the `**kwargs` approach. Updated, along with other suggestions.